### PR TITLE
qshell 2.1.3 (new formula)

### PR DESCRIPTION
**Pull request by jemygraw**

### CODE REVIEW

File: Formula/qshell.rb [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+  homepage "https://github.com/qiniu/qshell"
+  url "https://dn-devtools.qbox.me/qshell-2.1.3.tar.gz"
+  sha256 "26cad79300c471f34f52def329635f5405f520c7b662b6488a942173e1e93342"
+
+  def install
+    system "make", "install"
+  end
+
+  test do
+    system "qshell" "help"
```
Comment (bfontaine) [inline Formula/qshell.rb]: You missed a comma between `"qshell"` and `"help"`. Also, use `bin/"qshell"` instead.

File: Formula/qshell.rb [hunk 2]
```diff
@@ -0,0 +1,14 @@
+class Qshell < Formula
+  desc "Qiniu Cloud Resource Management Tool "
```
Comment (bfontaine) [inline Formula/qshell.rb]: Trailing space here.

File: Formula/qshell.rb [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+  homepage "https://github.com/qiniu/qshell"
+  url "https://dn-devtools.qbox.me/qshell-2.1.3.tar.gz"
```
Comment (bfontaine) [inline Formula/qshell.rb]: We need to build from source here.

File: Formula/qshell.rb [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+  sha256 "26cad79300c471f34f52def329635f5405f520c7b662b6488a942173e1e93342"
+
+  def install
+    system "make", "install"
```
Comment (bfontaine) [inline Formula/qshell.rb]: There’s no Makefile there.

**Review by bfontaine — CHANGES_REQUESTED** [review]

**Comment by fxcoudert** [comment]
**[Homebrew formulas must build from source](https://github.com/Homebrew/brew/blob/master/docs/Acceptable-Formulae.md)**

File: Formula/qshell.rb (hunk 3 above)
Comment (jemygraw) [inline Formula/qshell.rb]: if i want to publish a binary executable file, is there any docs i can refer to? many thanks

Comment (bfontaine) [inline Formula/qshell.rb]: The [same docs](https://docs.brew.sh/Formula-Cookbook.html) as other formulae work here too, but you’ll have to host the formula in [your own tap](https://docs.brew.sh/How-to-Create-and-Maintain-a-Tap.html) as Homebrew disallow pre-built binaries in the official taps.

In your case the only thing you need to change (beside the syntax issue in the `test do` block) is the `install` method:

```ruby
def install
  bin.install "qshell"
end
```

**Pull request closed without merging**